webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js

```diff
@@ -0,0 +1,194 @@
+
+  const [selectedViewBy, setSelectedViewBy] = useState('Different');
+  const viewBySelectionOptions = ['Different', 'Same'];
+
+  const clearSelectionVersionOne = () => {
+    setVersionOne(null);
+    setIsOpenSelectVersionOne(false);
+  };
+  const clearSelectionVersionTwo = () => {
+    setVersionTwo(null);
+    setIsOpenSelectVersionTwo(false);
+  };
+
+  const clearSelectionViewBy = () => {
+    setSelectedViewBy('');
+    setIsOpenSelectViewBy(false);
+  };
+  const onSelectVersionOne = (_event, selection, isPlaceholder) => {
+    if (isPlaceholder) {
+      clearSelectionVersionOne();
+    } else {
+      setVersionOne(selection);
+      setIsOpenSelectVersionOne(false);
+    }
+  };
+
+  const onSelectVersionTwo = (_event, selection, isPlaceholder) => {
+    if (isPlaceholder) {
+      clearSelectionVersionTwo();
+    } else {
+      setVersionTwo(selection);
+      setIsOpenSelectVersionTwo(false);
+    }
+  };
+
+  const onSelectViewBy = (_event, selection, isPlaceholder) => {
+    if (isPlaceholder) {
+      clearSelectionViewBy();
+    } else {
+      setSelectedViewBy(selection);
+      setIsOpenSelectViewBy(false);
+    }
+  };
+
+  const filteredVersions = selectedVersion =>
+    versions?.filter(result => Number(result.version) !== Number(selectedVersion));
+  const filteredVersionsFirstSelect = versions?.filter(result =>
+    Number(result.version) !== Number(versionTwo));
+  const filteredVersionsSecondSelect = versions?.filter(result =>
+    Number(result.version) !== Number(versionOne));
+  const selectionOptionsViewBy = viewBySelectionOptions.map(option => (
+    <SelectOption
+      key={option}
+      value={option}
+    >
+      {__(`${option}`)}
+    </SelectOption>
+  ));
+  return (
+    <Grid hasGutter style={{ margin: '0 24px' }}>
+      <GridItem span={12} style={{ display: 'flex' }}>
+        <TextContent>
+          <Text component={TextVariants.h2}>{__('Compare')}</Text>
+        </TextContent>
+        <Tooltip aria="none" aria-live="polite" content={toolTipContent} style={{ marginLeft: 'auto' }}>
+          <Button
+            aria-label="Clipboard"
+            variant="plain"
+            id="tt-ref"
+          >
+            <OutlinedQuestionCircleIcon />
+          </Button>
+        </Tooltip>
+      </GridItem>
+      <GridItem span={12}>
+        <Flex className="example-border">
+          <FlexItem>
+            <Flex direction={{ default: 'column' }}>
+              <TextContent>
+                <Text component={TextVariants.h4}>{__('Versions to compare')}</Text>
+              </TextContent>
+              <FlexItem>
+                <Flex className="example-border">
+                  <FlexItem>
+                    <Select
+                      style={{ marginRight: '10px' }}
+                      variant={SelectVariant.single}
+                      placeholderText="Select an option"
+                      aria-label="Select version one"
```

Comment:
Recommend adding `ouiaId` prop to this and all supported PF components.

Reply:
@jeremylenz would you mind clarifying what do you mean by "PF"? :smiley:

Reply:
sorry.. Patternfly components. The list of supported components is here: https://www.patternfly.org/v4/developer-resources/open-ui-automation#ouia-compliant-patternfly-4-components